src/commands/createWebApp/domainLabelScope/WebAppWithDomainLabelScopeCreateStep.ts

```diff
@@ -0,0 +1,217 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { type NameValuePair, type SiteConfig, type WebSiteManagementClient } from '@azure/arm-appservice';
+import { createHttpHeaders, createPipelineRequest } from '@azure/core-rest-pipeline';
+import { createWebSiteClient, WebsiteOS, type CustomLocation } from '@microsoft/vscode-azext-azureappservice';
+import { createGenericClient, LocationListStep, type AzExtPipelineResponse, type AzExtRequestPrepareOptions } from '@microsoft/vscode-azext-azureutils';
+import { AzureWizardExecuteStep } from '@microsoft/vscode-azext-utils';
+import { type AppResource } from '@microsoft/vscode-azext-utils/hostapi';
+import { type Progress } from 'vscode';
+import * as constants from '../../../constants';
+import { ext } from '../../../extensionVariables';
+import { localize } from '../../../localize';
+import { nonNullProp } from '../../../utils/nonNull';
+import { type FullJavaStack, type FullWebAppStack, type IWebAppWizardContext } from '../IWebAppWizardContext';
+import { getJavaLinuxRuntime } from '../stacks/getJavaLinuxRuntime';
+import { type WebAppStackValue, type WindowsJavaContainerSettings } from '../stacks/models/WebAppStackModel';
+import { type SitePayload } from './types';
+
+export class WebAppWithDomainLabelScopeCreateStep extends AzureWizardExecuteStep<IWebAppWizardContext> {
+    public priority: number = 140;
+
+    public async execute(context: IWebAppWizardContext, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
```

Comment:
Most of this file is a straight carryover of the `WebAppCreateStep` file, but I had to make a call manually without the SDK which means I added a `GenericClient` call to the API endpoint, and I had to slightly amend the site payload shape because it's different than when updating using the SDK